modules/light-clients/09-localhost/client_state.go

```diff
@@ -95,7 +91,17 @@ func (cs ClientState) VerifyMembership(
 	path exported.Path,
 	value []byte,
 ) error {
-	bz := store.Get([]byte(path.String()))
+	merklePath, ok := path.(commitmenttypes.MerklePath)
+	if !ok {
+		return sdkerrors.Wrapf(clienttypes.ErrFailedChannelStateVerification, "todo: update error -- not found for path %s", path)
+	}
+
+	// todo: revisit this
+	if len(merklePath.GetKeyPath()) != 2 && merklePath.KeyPath[0] != exported.StoreKey {
+		return sdkerrors.Wrapf(clienttypes.ErrFailedChannelStateVerification, "todo: update error -- not found for path %s", path)
+	}
+
+	bz := store.Get([]byte(merklePath.KeyPath[1]))
```

Comment:
could be a good candidate for an unexported function.
```go
func readPathBytes(path exported.Path,) ([]byte, error) {...}
```

Reply:
I think its more or less ok as is, given its a relatively small method compared to solomachine or tm